### PR TITLE
Add cities data

### LINKED ISSUE #271: Add Json file of all Canadian cities/towns/villages
Use this json file: https://github.com/jprichardson/node-canada/blob/master/data/cities.json and delete the cities that fails using Mapbox.

### PR DESCRIPTION
**Pull request by Shehryar21**
closes #271 

Worked very hard on this PR and spent countless hours writing around 7000 lines of code

### CODE REVIEW

**Review by nimwijetunga — APPROVED** [review]
LGTM but dinu should take a look as well before merging

**Review by DinuWije — APPROVED** [review]

**Pull request merged**
